pymbolic/geometric_algebra/mapper.py

```diff
@@ -232,13 +240,19 @@ def map_product(self, expr):
 
         # id to set((child index, axis), ...)
         nabla_finder = {}
+        has_d_source_nablas = False
+
+        for child_idx, child in enumerate(expr.children):
+            d_or_ns = self.derivative_collector(child)
+            if not d_or_ns:
+                d_source_nabla_ids_per_child.append(set())
+                continue
```

Comment:
Not quite sure if this is a valid check here, since I don't really understand how this nabla / source derivative / etc stuff works.

The intention was to catch when it doesn't find any derivatives and exit early.

Reply:
Seems fine.